File: sixtythreeebays_ui/src/services/ShippingService.js
import http from "../http-common"; 

class ShippingService {
  getAllShippings(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/shipping/shippings`, searchDTO);
  }

  get(shippingId) {
    return this.getRequest(`/shipping/${shippingId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/shipping?field=${matchData}`, null);
  }

  addShipping(data) {
    return http.post("/shipping/addShipping", data);
  }

  update(data) {
  	return http.post("/shipping/updateShipping", data);
  }
  
  uploadImage(data,shippingId) {
  	return http.postForm("/shipping/uploadImage/"+shippingId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ShippingService();

File: src/main/java/com/sixtythreeebays/service/ShippingService.java
package com.sixtythreeebays.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.sixtythreeebays.domain.Shipping;
import com.sixtythreeebays.dto.ShippingDTO;
import com.sixtythreeebays.dto.ShippingSearchDTO;
import com.sixtythreeebays.dto.ShippingPageDTO;
import com.sixtythreeebays.dto.ShippingConvertCriteriaDTO;
import com.sixtythreeebays.service.GenericService;
import com.sixtythreeebays.dto.common.RequestDTO;
import com.sixtythreeebays.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ShippingService extends GenericService<Shipping, Integer> {

	List<Shipping> findAll();

	ResultDTO addShipping(ShippingDTO shippingDTO, RequestDTO requestDTO);

	ResultDTO updateShipping(ShippingDTO shippingDTO, RequestDTO requestDTO);

    Page<Shipping> getAllShippings(Pageable pageable);

    Page<Shipping> getAllShippings(Specification<Shipping> spec, Pageable pageable);

	ResponseEntity<ShippingPageDTO> getShippings(ShippingSearchDTO shippingSearchDTO);
	
	List<ShippingDTO> convertShippingsToShippingDTOs(List<Shipping> shippings, ShippingConvertCriteriaDTO convertCriteria);

	ShippingDTO getShippingDTOById(Integer shippingId);







}





